src/main.rs

```diff
@@ -584,125 +608,133 @@ fn main() -> Result<(), Box<dyn Error>> {
                 .collect::<Result<Vec<_>,_>>()
         })?;
 
-    // Rare event sampling for pair creation
-    let pair_rate_increase = input.read::<f64,_>("control:increase_pair_rate_by")
-        // if increase is not specified at all, default to unity
-        .or_else(|e| match e.kind() {
-            InputErrorKind::Location => Ok(1.0),
-            _ => Err(e),
-        })
-        // failing that, check for automatic increase
-        .or_else(|e| match input.read::<String,_>("control:increase_pair_rate_by") {
-            Ok(s) if s == "auto" => if using_lcfa {
-                Ok(increase_lcfa_pair_rate_by(gamma, a0, wavelength))
-            } else {
-                Ok(increase_pair_rate_by(gamma, a0, wavelength, pol))
-            },
-            _ => Err(e),
-        })
-        .and_then(|r| if r < 1.0 {
-            eprintln!("Increase in pair creation rate must be >= 1.0.");
-            Err(InputError::conversion("control:increase_pair_rate_by", "increase_pair_rate_by"))
-        } else {
-            Ok(r)
-        })?;
-
-    let seed = 0x8658b90036b165ebu64 + ((rng_seed as u64) * 0x32f55cddaebae910u64);
-    let mut rng = Xoshiro256StarStar::seed_from_u64(seed);
-    for _i in 0..id {
-        rng.jump();
-    }
-
-    let nums: Vec<usize> = {
-        let tasks = ntasks as usize;
-        (0..tasks).map(|i| (npart * (i + 1) / tasks) - (npart * i / tasks)).collect()
-    };
-    assert_eq!(nums.iter().sum::<usize>(), npart);
-    let num = nums[id as usize];
-
-    if id == 0 {
-        println!("Running {} task{} with {} primary particles per task...", ntasks, if ntasks > 1 {"s"} else {""}, num);
-        #[cfg(feature = "with-mpi")] {
-            println!("\t* with MPI support enabled");
-        }
-        #[cfg(feature = "hdf5-output")] {
-            println!("\t* writing HDF5 output");
-        }
-        if pair_rate_increase > 1.0 {
-            println!("\t* with pair creation rate increased by {:.3e}", pair_rate_increase);
+    // Simulation building and running starts here
+    for a0_v in a0_values.iter() {
+        let a0: f64 = *a0_v; 
+        // Change to LCFA if a0 if variable is looped and LMA has been selected
+        if !using_lcfa && a0_values.len() > 1 {
+            if a0 > 20.0 {let using_lcfa = true;}
         }
```

Comment:
I understand why this is useful, but I'd prefer the code not to switch between LMA and LCFA unless asked to. My suggestion would be to remove this and I'll make a note to add something like `lcfa: auto` as an allowed input, which would do the switching if the collision parameters were out of bounds.